src/transaction/simple.rs

```diff
@@ -0,0 +1,271 @@
+            #[cfg(feature = "pure_rust")]
+            TypeConnectionContainer::PureRust(tr) => Ok(Transaction::new(tr)?.into()),
+        }
+    }
+
+    /// Commit the current transaction changes
+    pub fn commit(self) -> Result<(), FbError> {
+        match self.inner {
+            #[cfg(feature = "linking")]
+            TypeTransactionContainer::NativeDynLink(tr) => tr.commit(),
+            #[cfg(feature = "dynamic_loading")]
+            TypeTransactionContainer::NativeDynLoad(tr) => tr.commit(),
+            #[cfg(feature = "pure_rust")]
+            TypeTransactionContainer::PureRust(tr) => tr.commit(),
+        }
+    }
+
+    /// Commit the current transaction changes, but allowing to reuse the transaction
+    pub fn commit_retaining(&mut self) -> Result<(), FbError> {
+        match &mut self.inner {
+            #[cfg(feature = "linking")]
+            TypeTransactionContainer::NativeDynLink(tr) => tr.commit_retaining(),
+            #[cfg(feature = "dynamic_loading")]
+            TypeTransactionContainer::NativeDynLoad(tr) => tr.commit_retaining(),
+            #[cfg(feature = "pure_rust")]
+            TypeTransactionContainer::PureRust(tr) => tr.commit_retaining(),
+        }
+    }
+
+    /// Rollback the current transaction changes, but allowing to reuse the transaction
+    pub fn rollback_retaining(&mut self) -> Result<(), FbError> {
+        match &mut self.inner {
+            #[cfg(feature = "linking")]
+            TypeTransactionContainer::NativeDynLink(tr) => tr.rollback_retaining(),
+            #[cfg(feature = "dynamic_loading")]
+            TypeTransactionContainer::NativeDynLoad(tr) => tr.rollback_retaining(),
+            #[cfg(feature = "pure_rust")]
+            TypeTransactionContainer::PureRust(tr) => tr.rollback_retaining(),
+        }
+    }
+
+    /// Rollback the current transaction changes
+    pub fn rollback(self) -> Result<(), FbError> {
+        match self.inner {
+            #[cfg(feature = "linking")]
+            TypeTransactionContainer::NativeDynLink(tr) => tr.rollback(),
+            #[cfg(feature = "dynamic_loading")]
+            TypeTransactionContainer::NativeDynLoad(tr) => tr.rollback(),
+            #[cfg(feature = "pure_rust")]
+            TypeTransactionContainer::PureRust(tr) => tr.rollback(),
+        }
+    }
+
+    /// Execute the statement without returning any row
+    pub fn execute_immediate(&mut self, sql: &str) -> Result<(), FbError> {
+        match &mut self.inner {
+            #[cfg(feature = "linking")]
+            TypeTransactionContainer::NativeDynLink(tr) => tr.execute_immediate(sql),
+            #[cfg(feature = "dynamic_loading")]
+            TypeTransactionContainer::NativeDynLoad(tr) => tr.execute_immediate(sql),
+            #[cfg(feature = "pure_rust")]
+            TypeTransactionContainer::PureRust(tr) => tr.execute_immediate(sql),
+        }
+    }
+
+    // TODO: add the prepare() method
```

Comment:
not a blocker, since the old functionality is still present in `transaction/mod.rs`, but-
 is there some special challenge for implementing `prepare()` for `SimpleTransaction`?

seems like it would be "*simple*"

Reply:
No, it's not simple. To provide the `prepare` method we need implement a ´StatementSimple´ too. I think this is not a problem for now and if someone ask this, we can review and find another approach.